metricbeat/module/kafka/partition/partition.go

```diff
@@ -123,7 +121,7 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 
 				if !offOK {
 					if err == nil {
-						err = errFailQueryOffset
+						err = errors.Wrap(err, "operation failed")
```

Comment:
`err == nil` here, so `errors.Wrap(err, "operation failed")` will be also `nil`.

The case where I think we are not giving enough information is when one of the partition offsets are `-1` (and thus `offOk` is false).

I think we could get more information in `PartitionOffset`, it returns -1 if there is no offset for a specific topic and partition:
```
        block := resp.GetBlock(topic, partition)
        if len(block.Offsets) == 0 {
                return -1, nil
        }
```

But this `block` has also an `Err` attribute, maybe we can find there the cause, and we are not checking it. We should at least log at the debug level this error if not nil.

Reply:
And we could also log which one of the offsets (oldest or newest) is `-1`.

Reply:
It is an `int16`, but it [implements `Error()`](https://github.com/elastic/beats/blob/08bc430c3588e704afff14e41b9fbcb83008f143/vendor/github.com/Shopify/sarama/errors.go#L193) :slightly_smiling_face: so it gives an error message.